apps/server/src/modules/deletion/repo/mapper/deletion-log.mapper.spec.ts

```diff
@@ -0,0 +1,139 @@
+			});
+		});
+
+		describe('When entities array is mapped for domainObjects array', () => {
+			it('should properly map the entities to the domain objects', () => {
+				const entities = [deletionLogEntityFactory.build()];
+
+				const domainObjects = DeletionLogMapper.mapToDOs(entities);
+
+				const expectedDomainObjects = entities.map(
+					(entity) =>
+						new DeletionLog({
+							id: entity.id,
+							domain: entity.domain,
+							operation: entity.operation,
+							deletionRequestId: entity.deletionRequestId?.toHexString(),
+							modifiedCount: entity.modifiedCount,
+							deletedCount: entity.deletedCount,
+							createdAt: entity.createdAt,
+							updatedAt: entity.updatedAt,
+						})
+				);
+
+				expect(domainObjects).toEqual(expectedDomainObjects);
+			});
+		});
+	});
+
+	describe('mapToEntity', () => {
+		describe('When domainObject is mapped for entity', () => {
+			beforeAll(() => {
+				jest.useFakeTimers();
+				jest.setSystemTime(new Date());
+			});
+
+			afterAll(() => {
+				jest.useRealTimers();
+			});
+
+			it('should properly map the domainObject to the entity', () => {
+				const domainObject = deletionLogFactory.build();
+
+				const entities = DeletionLogMapper.mapToEntity(domainObject);
+
+				const expectedEntities = new DeletionLogEntity({
+					id: domainObject.id,
+					domain: domainObject.domain,
+					operation: domainObject.operation,
+					deletionRequestId: new ObjectId(domainObject.deletionRequestId),
+					modifiedCount: domainObject.modifiedCount,
+					deletedCount: domainObject.deletedCount,
+					createdAt: domainObject.createdAt,
+					updatedAt: domainObject.updatedAt,
+				});
+
+				expect(entities).toEqual(expectedEntities);
+			});
+		});
+	});
+
+	describe('mapToEntities', () => {
+		describe('When empty domainObjects array is mapped for an entities array', () => {
+			it('should return empty entities array for an empty domain objects array', () => {
+				const entities = DeletionLogMapper.mapToEntities([]);
+
+				expect(entities).toEqual([]);
+			});
+		});
+
+		describe('When domainObjects array is mapped for entities array', () => {
+			beforeAll(() => {
+				jest.useFakeTimers();
+				jest.setSystemTime(new Date());
+			});
+
+			afterAll(() => {
+				jest.useRealTimers();
+			});
+
+			it('should properly map the domainObjects to the entities', () => {
+				const domainObjects = [deletionLogFactory.build()];
+
+				const entities = DeletionLogMapper.mapToEntities(domainObjects);
+
+				const expectedEntities = domainObjects.map(
+					(domainObject) =>
+						new DeletionLogEntity({
+							id: domainObject.id,
+							domain: domainObject.domain,
+							operation: domainObject.operation,
+							deletionRequestId: new ObjectId(domainObject.deletionRequestId),
+							modifiedCount: domainObject.modifiedCount,
+							deletedCount: domainObject.deletedCount,
+							createdAt: domainObject.createdAt,
+							updatedAt: domainObject.updatedAt,
+						})
+				);
```

Comment:
Please move it (as well as the entity preparation in the first place) to the `setup()` function.